CalibMuon/DTCalibration/plugins/DTVDriftWriter.cc

```diff
@@ -32,22 +34,38 @@ using namespace edm;
 
 DTVDriftWriter::DTVDriftWriter(const ParameterSet& pset)
     : granularity_(pset.getUntrackedParameter<string>("calibGranularity", "bySL")),
+      mTimeMap_(nullptr),
+      vDriftMap_(nullptr),
       vDriftAlgo_{DTVDriftPluginFactory::get()->create(pset.getParameter<string>("vDriftAlgo"),
                                                        pset.getParameter<ParameterSet>("vDriftAlgoConfig"))} {
   LogVerbatim("Calibration") << "[DTVDriftWriter]Constructor called!";
 
   if (granularity_ != "bySL")
     throw cms::Exception("Configuration")
         << "[DTVDriftWriter] Check parameter calibGranularity: " << granularity_ << " option not available.";
+
+  readLegacyVDriftDB = pset.getParameter<bool>("readLegacyVDriftDB");
+  writeLegacyVDriftDB = pset.getParameter<bool>("writeLegacyVDriftDB");
 }
 
 DTVDriftWriter::~DTVDriftWriter() { LogVerbatim("Calibration") << "[DTVDriftWriter]Destructor called!"; }
 
 void DTVDriftWriter::beginRun(const edm::Run& run, const edm::EventSetup& setup) {
-  // Get the map of ttrig from the Setup
-  ESHandle<DTMtime> mTime;
-  setup.get<DTMtimeRcd>().get(mTime);
-  mTimeMap_ = &*mTime;
+  // Get the map of vdrift from the Setup
+  if (readLegacyVDriftDB) {
+    ESHandle<DTMtime> mTime;
+    setup.get<DTMtimeRcd>().get(mTime);
+    mTimeMap_ = &*mTime;
+  } else {
+    ESHandle<DTRecoConditions> hVdrift;
+    setup.get<DTRecoConditionsVdriftRcd>().get(hVdrift);
+    vDriftMap_ = &*hVdrift;
+    // Consistency check: no parametrization is implemented for the time being
+    int version = vDriftMap_->version();
+    if (version != 1) {
+      throw cms::Exception("Configuration") << "only version 1 is presently supported for VDriftDB";
+    }
+  }
 
   // Get geometry from Event Setup
   setup.get<MuonGeometryRecord>().get(dtGeom_);
```

Comment:
here

Reply:
Besidesm the same migration would have to be done everywhere in the package; for example. also in 
https://github.com/cms-sw/cmssw/blob/master/CalibMuon/DTCalibration/plugins/DTTTrigCorrection.cc#L49
an generally in each and every cc under plugins, not just in those interested by this PR.
IMO this is another reason why that migration deserves its own, separate PR.
@silviodonato can you please let me know if you agree.